tests/functional/base_metric_types/test_base_average_metric.py

```diff
@@ -0,0 +1,107 @@
+from struct import pack
+import os
+import pytest
+from dbt.tests.util import run_dbt
+
+# our file contents
+from tests.functional.fixtures import (
+    fact_orders_source_csv,
+    fact_orders_sql,
+    fact_orders_yml,
+)
+
+# models/base_average_metric.sql
+base_average_metric_sql = """
+select *
+from 
+{{ metrics.calculate(metric('base_average_metric'), 
+    grain='month'
+    )
+}}
+"""
+
+# models/base_average_metric.yml
+base_average_metric_yml = """
+version: 2 
+models:
+  - name: base_average_metric
+    tests: 
+      - dbt_utils.equality:
+          compare_model: ref('base_average_metric__expected')
+metrics:
+  - name: base_average_metric
+    model: ref('fact_orders')
+    label: Total Discount ($)
+    timestamp: order_date
+    time_grains: [day, week, month]
+    type: average
+    sql: discount_total
+    dimensions:
+      - had_discount
+      - order_country
+"""
+
+# seeds/base_average_metric__expected.csv
+base_average_metric__expected_csv = """
+date_month,base_average_metric
+2022-01-01,1.00000000000000000000
+2022-02-01,1.3333333333333333
+""".lstrip()
```

Comment:
Okkkkkk so am I understanding this correctly: 

These lines of hardcoded logic are the same as the ones that are up above as proper .sql/.yml/.csv files? So we're duplicating everything? How do they stay in sync? And why do we need the former if it's all just turning into string constants down here? 

idk if this is pythonic or not, but I would strongly prefer something like 

```python
#https://stackoverflow.com/a/49564464/14007029
from pathlib import Path
base_average_metric_sql = Path('../models/base_average_metric.sql').read_text();
base_average_metric_yml = Path('../models/base_average_metric.yml').read_text();
```

Such that we don't have to redefine everything from scratch in two places 🤢

Reply:
Not quite. There are static datasets/definitions contained in `fixtures.py` and those are then imported. The reason that I keep some components outside of it (like the definition of the metric) is so we can understand what metric is being tested while just looking at the single file. 

If we're trying to be more pythonic then I should move that definition into the fixtures file.

Reply:
OK! I think I was caught up on thinking that the stuff in the integration tests project was this stuff's twin. Carry on!